jdaviz/configs/default/plugins/metadata_viewer/metadata_viewer.py

```diff
@@ -34,13 +43,51 @@ def _show_metadata(self, event):
         else:
             meta = data.meta
 
+        if PRIHDR_KEY in data.meta:
+            self.has_primary = True
+        else:
+            self.has_primary = False
+
+        self.show_primary = False
+        self.find_public_metadata(meta, primary_only=False)
+
+    @observe("show_primary")
+    def handle_show_primary(self, event):
+        if not self.show_primary:
+            self.show_metadata(event)
+            return
+
+        data = self.dataset.selected_dc_item
+        if (data is None or not hasattr(data, 'meta') or not isinstance(data.meta, dict)
+                or len(data.meta) < 1):
+            self.reset()
+            return
+
+        self.find_public_metadata(data.meta, primary_only=True)
+
+    def find_public_metadata(self, meta, primary_only=False):
+        if primary_only:
+            if PRIHDR_KEY in meta:
+                meta = meta[PRIHDR_KEY]
+            else:
+                return []
+
         d = flatten_nested_dict(meta)
-        for badkey in ('COMMENT', 'HISTORY', ''):
+        # Some FITS keywords cause "# ipykernel cannot clean for JSON" messages.
+        # Also, we want to hide internal metadata that starts with underscore.
+        badkeys = ['COMMENT', 'HISTORY', ''] + [k for k in d if k.startswith('_')]

```

Comment:
@kecnry , I think we talked about automatically hiding "hidden" metadata? This would do it for you.